doc/embed_wamr.md

```diff
@@ -258,68 +258,98 @@ We can't pass structure data or class objects through the pointer since the memo
 
 ## Execute wasm functions in multiple threads
 
-The `exec_env` is not thread safety, it will cause unexpected behavior if the same `exec_env` is used in multiple threads. However, we've provided two ways to execute wasm functions concurrently:
+It isn't safe to use an `exec_env` object in multiple threads concurrently.
+To run a multi-threaded application, you basically need a separate `exec_env`
+for each threads.
 
-- You can use `pthread` APIs in your wasm application, see [pthread library](./pthread_library.md) for more details.
+### Approaches to manage `exec_env` objects and threads
 
-- The `spawn exec_env` and `spawn thread` APIs are available, you can use these APIs to manage the threads in native:
+WAMR supports two approaches to manage `exec_env` and threads as described
```

Comment:
Isn't wasi's spawn thread the third one?

Reply:
"two" here means:

1. pthread (lib-pthread or wasi-threads)
2. wasm_runtime_spawn_exec_env/wasm_runtime_spawn_thread

Reply:
I see. I wouldn't call `wasi-threads` a `pthread` - `pthread` is just one of the (possibly) many wrappers around `wasi-threads` (even though pthreads is currently the only implementation that wraps this hostcall). I was a bit confused by the line 273; given that `pthread` library is deprecated, we might point to `wasi-libc` instead

Reply:
ok. i tweaked wording a bit.